apis/v1alpha1/instrumentation_webhook_test.go

```diff
@@ -18,24 +20,525 @@ import (
 var defaultVolumeSize = resource.MustParse("200Mi")
 
 func TestInstrumentationDefaultingWebhook(t *testing.T) {
-	inst := &Instrumentation{}
-	err := InstrumentationWebhook{
-		cfg: config.New(
-			config.WithAutoInstrumentationJavaImage("java-img:1"),
-			config.WithAutoInstrumentationNodeJSImage("nodejs-img:1"),
-			config.WithAutoInstrumentationPythonImage("python-img:1"),
-			config.WithAutoInstrumentationDotNetImage("dotnet-img:1"),
-			config.WithAutoInstrumentationApacheHttpdImage("apache-httpd-img:1"),
-			config.WithAutoInstrumentationNginxImage("nginx-img:1"),
-		),
-	}.Default(context.Background(), inst)
-	assert.NoError(t, err)
-	assert.Equal(t, "java-img:1", inst.Spec.Java.Image)
-	assert.Equal(t, "nodejs-img:1", inst.Spec.NodeJS.Image)
-	assert.Equal(t, "python-img:1", inst.Spec.Python.Image)
-	assert.Equal(t, "dotnet-img:1", inst.Spec.DotNet.Image)
-	assert.Equal(t, "apache-httpd-img:1", inst.Spec.ApacheHttpd.Image)
-	assert.Equal(t, "nginx-img:1", inst.Spec.Nginx.Image)
+	tests := []struct {
```

Comment:
I think I preferred the way this test was written previously. You're right to want to add tests for other cases, but I'd prefer these to be plain subtests with unique assertions. Then you can have a test to check the resource override, and only assert that the override applies, instead of needing to include the whole instrumentation spec as an expected return value. Does that make sense?

Reply:
@swiatekm Thank you for your feedback! I've fixed the PR so could you take a look at it again when you have time?